dvc/stage/run.py

```diff
@@ -12,27 +14,58 @@
 logger = logger.getChild(__name__)
 
 
-def _make_cmd(executable, cmd):
-    if executable is None:
-        return cmd
-    opts = {"zsh": ["--no-rcs"], "bash": ["--noprofile", "--norc"]}
-    name = os.path.basename(executable).lower()
-    return [executable, *opts.get(name, []), "-c", cmd]
-
+def _fish_supports_no_config(executable) -> bool:
+    """
+    Check if the fish shell supports the --no-config option.
 
-def warn_if_fish(executable):
-    if executable is None or os.path.basename(os.path.realpath(executable)) != "fish":
-        return
+    Parameters:
+    executable (str): The path to the fish shell executable.
 
+    Returns:
+    bool: True if the fish version is greater than 3.3.0, False otherwise.
+    """
+    try:
+        output = subprocess.check_output(
+            [executable, "-c", "echo $FISH_VERSION"],  # noqa: S603
+            text=True,
+        )
+    except subprocess.CalledProcessError:
+        logger.warning("could not check fish version, defaulting to False")
+        return False
+    version = Version(output.strip())
+    version_to_check = Version("3.3.0")
+    return version > version_to_check
+
+
+def _warn_if_fish():
     logger.warning(
-        "DVC detected that you are using fish as your default "
-        "shell. Be aware that it might cause problems by overwriting "
+        "DVC detected that you are using a version of fish shell below 3.3.0 "
+        "Be aware that it might cause problems by overwriting "
         "your current environment variables with values defined "
-        "in '.fishrc', which might affect your command. See "
+        "in 'config.fish', which might affect your command. See "
```

Comment:
was it a bug with `.fishrc`, why renaming it here to `config.fish`?

Reply:
yup, there is no `.fishrc`. The configs are set in `config.fish`.